gcsfs/core.py

```diff
@@ -1060,13 +1061,32 @@ async def _ls(
             for entry in await self._list_objects(
                 path, prefix=prefix, versions=versions, **kwargs
             ):
+                if entry["size"] == 0 and entry["name"].endswith("/"):
+                    entry = {
+                        "bucket": entry["bucket"],
+                        "name": path.rstrip("/"),
+                        "size": 0,
+                        "storageClass": "DIRECTORY",
+                        "type": "directory",
+                    }
+                    if entry in out:
+                        continue
+
                 if versions and "generation" in entry:
                     entry = entry.copy()
                     entry["name"] = f"{entry['name']}#{entry['generation']}"
+
                 out.append(entry)
 
         if detail:
-            return out
+            return sorted(
+                out,
+                key=lambda e: (
```

Comment:
You want to show directories first? I think that's not the case in any other fsspec ls, it should be simple sorting.

Reply:
You previously commented that ´info(path)´ and ´ls(path)[0]´ should be equal.
Showing the directories first was an attempt at that insuring that.
But that’s not the case anyway, when a file shadows a “directory". If you have the files foo/bas and foo/bas/ ´info(path)´ shows type=file and´ls(path)[0]´ shows type=directory
Ordering by name is perhaps the simple and correct solution here.